static/app/views/performance/database/noDataMessage.tsx

```diff
@@ -1,106 +1,117 @@
+import {useOutdatedSDKProjects} from 'sentry/views/performance/database/useOutdatedSDKProjects';
 
 interface Props {
   Wrapper?: React.ComponentType;
+  isDataAvailable?: boolean;
 }
 
 function DivWrapper(props) {
   return <div {...props} />;
 }
 
-export function NoDataMessage({Wrapper = DivWrapper}: Props) {
+export function NoDataMessage({Wrapper = DivWrapper, isDataAvailable}: Props) {
   const {selection, isReady: pageFilterIsReady} = usePageFilters();
 
   const selectedProjectIds = selection.projects.map(projectId => projectId.toString());
 
-  const {data: projectSpanMetricsCounts, isLoading: areSpanMetricCountsLoading} =
-    useProjectSpanMetricCounts({
-      query: 'span.module:db',
-      statsPeriod: SAMPLE_STATS_PERIOD,
-      enabled: pageFilterIsReady,
+  const {projects: outdatedProjects, isFetching: areOutdatedProjectsFetching} =
+    useOutdatedSDKProjects({
       projectId: selectedProjectIds,
+      enabled: pageFilterIsReady && !isDataAvailable,
     });
 
-  const doesAnySelectedProjectHaveMetrics =
-    sumBy(projectSpanMetricsCounts, 'count()') > 0;
-
-  const {ineligibleProjects, isFetching: areIneligibleProjectsFetching} =
-    useIneligibleProjects({
+  const {projects: denylistedProjects, isFetching: areDenylistProjectsFetching} =
+    useDenylistedProjects({
       projectId: selectedProjectIds,
-      enabled: pageFilterIsReady && !doesAnySelectedProjectHaveMetrics,
     });
 
-  const organization = useOrganization();
-
-  const hasMoreIneligibleProjectsThanVisible =
-    ineligibleProjects.length > MAX_LISTED_PROJECTS;
-
-  if (areSpanMetricCountsLoading || areIneligibleProjectsFetching) {
+  // Hide message while fetching outdated projects or denylisted projects
+  if (areOutdatedProjectsFetching || areDenylistProjectsFetching) {
     return null;
   }
 
-  if (!projectSpanMetricsCounts) {
+  // Hide message if all conditions are satisfied
+  if (
+    isDataAvailable &&
+    !areOutdatedProjectsFetching &&
+    !areDenylistProjectsFetching &&
+    outdatedProjects.length === 0 &&
+    denylistedProjects.length === 0
+  ) {
     return null;
   }
 
-  if (doesAnySelectedProjectHaveMetrics) {
-    return null;
-  }
+  return (
+    <Wrapper>
+      {!isDataAvailable &&
+        tct(
+          'No queries found. Try updating your filters, or learn more about performance monitoring for queries in our [documentation:documentation].',
+          {
+            documentation: (
+              <ExternalLink href="https://docs.sentry.io/product/performance/queries/" />
+            ),
+          }
+        )}{' '}
+      {outdatedProjects.length > 0 &&
+        tct('You may be missing data due to outdated SDKs: [projectList].', {
+          projectList: <ProjectList projects={outdatedProjects} />,
+        })}{' '}
+      {denylistedProjects.length > 0 &&
+        tct(
+          'Some of your projects have been omitted from database metrics extraction. Please contact [support]. Omitted projects: [projectList].',
```

Comment:
l: Ideally we can avoid talking about metrics in general.
```suggestion
          'Some of your projects have been omitted from query metrics calculations. Please contact [support]. Omitted projects: [projectList].',
```

Reply:
I agree but I'm confused because your suggestion still mentions metrics!

Reply:
Sorry should have made it clearer that my suggestion is what I think we should do ideally (since it still mentions metrics), but at least avoids having to bring up extraction.

If we can find a way to remove mentioning metrics all together even better. 

Maybe something like `query performance analysis`, which matches some of the vernacular in the docs.

```suggestion
          'Some of your projects have been omitted from query performance analysis. Please contact [support]. Omitted projects: [projectList].',
```

Reply:
Works for me!